azkaban-common/src/main/java/azkaban/executor/ExecutionFlowDBManager.java

```diff
@@ -0,0 +1,302 @@
+
+    final String json = JSONUtils.toJSON(flow.toObject());
+    byte[] data = null;
+    try {
+      final byte[] stringData = json.getBytes("UTF-8");
+      data = stringData;
+      // Todo kunkun-tang: use a common method to transform stringData to data.
+      if (encType == EncodingType.GZIP) {
+        data = GZIPUtils.gzipBytes(stringData);
+      }
+    } catch (final IOException e) {
+      throw new ExecutorManagerException("Error encoding the execution flow.");
+    }
+
+    try {
+      this.dbOperator.update(UPDATE_EXECUTABLE_FLOW_DATA, flow.getStatus()
+          .getNumVal(), flow.getUpdateTime(), flow.getStartTime(), flow
+          .getEndTime(), encType.getNumVal(), data, flow.getExecutionId());
+    } catch (final SQLException e) {
+      throw new ExecutorManagerException("Error updating flow.", e);
+    }
+  }
+
+  public ExecutableFlow fetchExecutableFlow(final int execId) throws ExecutorManagerException {
+    final FetchExecutableFlows flowHandler = new FetchExecutableFlows();
+    try {
+      final List<ExecutableFlow> properties = this.dbOperator
+          .query(FetchExecutableFlows.FETCH_EXECUTABLE_FLOW, flowHandler, execId);
+      if (properties.isEmpty()) {
+        return null;
+      } else {
+        return properties.get(0);
+      }
+    } catch (final SQLException e) {
+      throw new ExecutorManagerException("Error fetching flow id " + execId, e);
+    }
+  }
+
+  public static class FetchExecutableFlows implements
+      ResultSetHandler<List<ExecutableFlow>> {
+
+    static String FETCH_BASE_EXECUTABLE_FLOW_QUERY =
+        "SELECT exec_id, enc_type, flow_data FROM execution_flows ";
+    static String FETCH_EXECUTABLE_FLOW =
+        "SELECT exec_id, enc_type, flow_data FROM execution_flows "
+            + "WHERE exec_id=?";
+    static String FETCH_ALL_EXECUTABLE_FLOW_HISTORY =
+        "SELECT exec_id, enc_type, flow_data FROM execution_flows "
+            + "ORDER BY exec_id DESC LIMIT ?, ?";
+    static String FETCH_EXECUTABLE_FLOW_HISTORY =
+        "SELECT exec_id, enc_type, flow_data FROM execution_flows "
+            + "WHERE project_id=? AND flow_id=? "
+            + "ORDER BY exec_id DESC LIMIT ?, ?";
+    static String FETCH_EXECUTABLE_FLOW_BY_STATUS =
+        "SELECT exec_id, enc_type, flow_data FROM execution_flows "
+            + "WHERE project_id=? AND flow_id=? AND status=? "
+            + "ORDER BY exec_id DESC LIMIT ?, ?";
+
+    @Override
+    public List<ExecutableFlow> handle(final ResultSet rs) throws SQLException {
+      if (!rs.next()) {
+        return Collections.emptyList();
+      }
+
+      final List<ExecutableFlow> execFlows = new ArrayList<>();
+      do {
+        final int id = rs.getInt(1);
+        final int encodingType = rs.getInt(2);
+        final byte[] data = rs.getBytes(3);
+
+        if (data != null) {
+          final EncodingType encType = EncodingType.fromInteger(encodingType);
+          final Object flowObj;
+          try {
```

Comment:
Just add a note here:
This part of the code is duplicate in multiple places in the code base now. When you do future refactoring, can you also improve it?